### PR TITLE
#3 Bug fix in the option parser and little improvement. Critical bug fix…

### PR DESCRIPTION
**Pull request by andressbarajas**
… of GNU Debugger (GDB) remote interface for MinGW/MSYS.

Pull request number 3 from Sizious dcload-serial Repo. Contains commits:

https://github.com/sizious/dcload-serial/commit/2814d5b1af20c038f71358302af438750116cffa

and

https://github.com/sizious/dcload-serial/commit/1677b5e0f9a11cf598807a622e6ea91bd10b3152

### CODE REVIEW

File: host-src/tool/dc-tool.c [hunk 1]
```diff
@@ -284,7 +299,11 @@ char serial_getc()
     retval = serial_read(&tmp, 1);
     if (retval == -1) {
         printf("serial_getc: read error!\n");
-        tmp = 0x00;
+#ifndef __MINGW32__		
+		tmp = (char)NULL;
+#else
+		tmp = 0x00;
+#endif
```
Comment (ljsebald) [inline host-src/tool/dc-tool.c]: This change shouldn't be needed. I'm pretty sure this is undoing something that was done in one of your earlier PRs.

**Comment by ljsebald** [comment]
There are several instances of changes using tabs instead of spaces for indentation. Please correct.

File: host-src/tool/dc-tool.c (hunk 1 above)
Comment (andressbarajas) [reply host-src/tool/dc-tool.c]: Fixed.

**Comment by andressbarajas** [comment]
I fixed spacing in dc-tool and syscalls

File: host-src/tool/syscalls.c [hunk 2]
```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
...
-	if ( (socket_fd = accept( gdb_server_socket, NULL, NULL )) < 0) {
-	    perror("error accepting gdb server connection");
-	    send_uint(-1);
-	    return;
-	}
+    if (socket_fd == 0) {
+        printf( "waiting for gdb client connection...\n" );
+        socket_fd = accept( gdb_server_socket, NULL, NULL );
+#ifdef __MINGW32__
+        if (socket_fd != INVALID_SOCKET)
```
Comment (ljsebald) [inline host-src/tool/syscalls.c]: Is this line correct? It just doesn't seem like it is...

File: host-src/tool/syscalls.c [hunk 3]
```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
...
+        }
     }
 
     if (in_size)
-	write(socket_fd, gdb_buf, in_size);
+#ifdef __MINGW32__
+        send(socket_fd, gdb_buf, in_size, 0);		
+#else		
+        write(socket_fd, gdb_buf, in_size);		
+#endif
```
Comment (ljsebald) [inline host-src/tool/syscalls.c]: I suspect that this could use send all around, rather than write...

File: host-src/tool/syscalls.c [hunk 4]
```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
...
 
     if (out_size) {
-	retval = read(socket_fd, gdb_buf, out_size > GDBBUFSIZE ? GDBBUFSIZE : out_size);
-	if (retval == 0)
-	    socket_fd = -1;
+#ifdef __MINGW32__
+        retval = recv(socket_fd, gdb_buf, out_size > GDBBUFSIZE ? GDBBUFSIZE : out_size, 0);
+#else
+        retval = read(socket_fd, gdb_buf, out_size > GDBBUFSIZE ? GDBBUFSIZE : out_size);
+#endif
```
Comment (ljsebald) [inline host-src/tool/syscalls.c]: Same goes here. I don't think there's a reason why recv wouldn't work everywhere.

File: host-src/tool/syscalls.c (hunk 3 above)
Comment (andressbarajas) [reply host-src/tool/syscalls.c:518]: @sizious

File: host-src/tool/syscalls.c (hunk 2 above)
Comment (andressbarajas) [reply host-src/tool/syscalls.c]: @sizious

File: host-src/tool/syscalls.c (hunk 4 above)
Comment (andressbarajas) [reply host-src/tool/syscalls.c:525]: @sizious

Comment (andressbarajas) [reply host-src/tool/syscalls.c]: Now uses recv all around

File: host-src/tool/syscalls.c (hunk 3 above)
Comment (andressbarajas) [reply host-src/tool/syscalls.c]: Now uses send all around

File: host-src/tool/syscalls.c (hunk 2 above)
Comment (andressbarajas) [reply host-src/tool/syscalls.c]: Fixed

**Pull request merged**